readium/shared/src/main/java/org/readium/r2/shared/util/http/HttpResourceFactory.kt

```diff
@@ -8,14 +8,27 @@ package org.readium.r2.shared.util.http
 
 import org.readium.r2.shared.util.AbsoluteUrl
 import org.readium.r2.shared.util.Try
+import org.readium.r2.shared.util.mediatype.MediaType
 import org.readium.r2.shared.util.resource.Resource
 import org.readium.r2.shared.util.resource.ResourceFactory
 
 public class HttpResourceFactory(
     private val httpClient: HttpClient
 ) : ResourceFactory {
 
-    override suspend fun create(url: AbsoluteUrl): Try<Resource, ResourceFactory.Error> {
+    override suspend fun create(url: AbsoluteUrl): Resource? {
+        if (!url.isHttp) {
+            return null
+        }
+
+        // FIXME: should make a head request to check that url points to a resource
```

Comment:
Is it to account for the exploded directory served through HTTP? It can get complicated to infer as you could get an HTML page back but still have an exploded directory. Even checking for a trailing `/` in the URL is not foolproof.


IMHO an acceptable alternative would be to only support remote exploded publication if they expose a RWPM. A "raw" exploded directory is currently not a use case.